src/main/kotlin/org/rust/ide/surroundWith/RustWithLoopSurrounder.kt

```diff
@@ -0,0 +1,28 @@
+package org.rust.ide.surroundWith
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RustElementFactory
+import org.rust.lang.core.psi.RustLoopExprElement
+
+class RustWithLoopSurrounder: RustStatementsSurrounderBase() {
+    override fun getTemplateDescription(): String = "loop { }"
+
+    override fun surroundStatements(
+        project: Project,
+        editor: Editor,
+        container: PsiElement,
+        statements: Array<out PsiElement>
+    ): TextRange? {
+        var loopStmt = RustElementFactory.createExpression(project, "loop {\n}") as RustLoopExprElement
```

Comment:
It's usually better to avoid `var`, though it does not matter much in this case.